src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/blocks/rightcurly/InputRightCurlyCaseBlocksWithSwitchExpressionAloneOrSingleline.java

```diff
@@ -0,0 +1,93 @@
+/*
+RightCurly
+option = ALONE_OR_SINGLELINE
+tokens = LITERAL_CASE, LITERAL_IF
+
+*/
+
+//non-compiled with javac: Compilable with Java14
+package com.puppycrawl.tools.checkstyle.checks.blocks.rightcurly;
+
+public class InputRightCurlyCaseBlocksWithSwitchExpressionAloneOrSingleline {
+
+    int foo(int yield) {
+        return switch (yield) {
+            case 1 -> 2;
+            case 2 -> 3;
+            case 3 -> 4;
+            default -> 5;
+        };
+    }
+
+    void test() {
+        int x = 1;
+        String s = switch (x) {
+            case 1 -> {
+
+                yield "and that's it";
+            }  case 2 -> "x is 2";
+            default -> "x is neither 1 nor 2";
+        };
+    }
+
+    void test2() {
+        int x = 1;
+        String s = switch (x) {
+            case 1 -> {
+
+                yield "and that's it";
+            }  case 2 -> { yield "x is 2";}
+            default -> "x is neither 1 nor 2";
+        };
+    }
+
+    void test3() {
+        int x = 1;
+         String s = switch (x) {
+            case 1 -> {
+
+                String s1 = "and that's it";
+                yield s1;
+            } case 2 -> { yield "x is 2";}
+            default -> "x is neither 1 nor 2";
+        };
+    }
+
+    void test4() throws Exception {
+           int a = 0;
+           int b = switch (a) {
+                case 0 -> {
+                    int x = 5;
+                    int y = 6;
+                    if (a == 2) {
+                        y = 7;}   // violation '}' at column 31 should be alone on a line'
+                    throw new Exception();
+                }  default -> 2;
```

Comment:
Please confirm if this token validates this type of line. I noticed there is a violation on another file in a similar manner, but there isn't one here as it is not alone or single line.

https://github.com/checkstyle/checkstyle/pull/14623/files#diff-01f1a1b1234b46ef06ab0849a66bdb1efd5d8cdb080408432bd308213423a598R32

Reply:
> but there isn't one here as it is not alone or single line.

there is no violation here because this is switch expression and we are supposed to ignore it . the other example is normal switch rule

Reply:
So `case` includes switch statements (using `:` and `->`) but not switch expressions where the switch is used in said expression?

Reply:
as per my understanding In a switch expression, rather than just executing code blocks, the switch itself produces a value, which can then be assigned to a variable or directly used in an expression.
ex:
```
sout(switch(...) )
int a = switch () {...}
return switch (){....}
```
if switch doesn't produce a value to be used and it just executes code blocks then it is switch statement or rule
Check [JEP](https://openjdk.org/jeps/361) for more info